keras_cv/layers/preprocessing3d/random_pasting_objects.py

```diff
@@ -0,0 +1,242 @@
+        self,
+        point_clouds,
+        bounding_boxes,
+        object_point_clouds,
+        object_bounding_boxes,
+        **kwargs
+    ):
+        del point_clouds
+        num_paste_bounding_boxes = self._random_generator.random_uniform(
+            (),
+            minval=self._min_paste_bounding_boxes,
+            maxval=self._max_paste_bounding_boxes,
+        )
+        num_paste_bounding_boxes = tf.cast(num_paste_bounding_boxes, dtype=tf.int32)
+        num_existing_bounding_boxes = tf.shape(bounding_boxes)[1]
+        if self._label_index:
+            object_mask = tf.math.equal(
+                object_bounding_boxes[0, :, 7], self._label_index
+            )
+            object_point_clouds = tf.boolean_mask(
+                object_point_clouds, object_mask, axis=1
+            )
+            object_bounding_boxes = tf.boolean_mask(
+                object_bounding_boxes, object_mask, axis=1
+            )
+        shuffle_index = tf.range(tf.shape(object_point_clouds)[1])
+        shuffle_index = tf.random.shuffle(shuffle_index)
+        object_point_clouds = tf.gather(object_point_clouds, shuffle_index, axis=1)
+        object_bounding_boxes = tf.gather(object_bounding_boxes, shuffle_index, axis=1)
+
+        # Load at most 5 times num_paste_bounding_boxes to check overlaps.
+        num_compare_boounding_boxes = tf.math.minimum(
+            num_paste_bounding_boxes * 5,
+            tf.shape(object_point_clouds)[1],
+        )
+
+        object_point_clouds = object_point_clouds[:, :num_compare_boounding_boxes, :]
+        object_bounding_boxes = object_bounding_boxes[
+            :, :num_compare_boounding_boxes, :
+        ]
+        # Use the current frame to check overlap between pasted bounding boxes and existing bounding boxes
+        all_bounding_boxes = tf.concat([bounding_boxes, object_bounding_boxes], axis=1)
+        iou = self._iou_3d(all_bounding_boxes[0, :, :7], all_bounding_boxes[0, :, :7])
+        iou = tf.linalg.band_part(iou, -1, 0)
+        iou_sum = tf.reduce_sum(iou[num_existing_bounding_boxes:], axis=1)
+        # A non overlapping bounding box has a 1.0 IoU with itself.
+        non_overlapping_mask = tf.reshape(iou_sum <= 1, [-1])
+        object_point_clouds = tf.boolean_mask(
+            object_point_clouds, non_overlapping_mask, axis=1
+        )
+        object_bounding_boxes = tf.boolean_mask(
+            object_bounding_boxes, non_overlapping_mask, axis=1
+        )
+        object_point_clouds = object_point_clouds[:, :num_paste_bounding_boxes, :]
+        object_bounding_boxes = object_bounding_boxes[:, :num_paste_bounding_boxes, :]
+        return {
+            "object_point_clouds": object_point_clouds,
+            "object_bounding_boxes": object_bounding_boxes,
+        }
+
+    def augment_point_clouds_bounding_boxes(
+        self, point_clouds, bounding_boxes, transformation, **kwargs
+    ):
+        object_point_clouds = transformation["object_point_clouds"]
```

Comment:
nit: is it weird to name this "object_point_clouds", given this is actually a different set of batch samples (batch size - effective batch size) with "point_clouds" (effective batch size)?

Reply:
How about addition_object_point_clouds?